test/controllers/declaration/ItemTypePageControllerSpec.scala

```diff
@@ -569,15 +775,25 @@ object ItemTypePageControllerSpec {
     cusCode: String = "",
     statisticalValue: String = ""
   ): Map[String, String] =
-    Map(formAction.toString -> "") ++
+    addActionTypeToFormData(
+      formAction,
       Map(
         "combinedNomenclatureCode" -> combinedNomenclatureCode,
         "descriptionOfGoods" -> descriptionOfGoods,
         "cusCode" -> cusCode,
         "statisticalValue" -> statisticalValue
       ) ++ taricAdditionalCodes.zipWithIndex.map {
-      case (code, idx) => taricAdditionalCodesKey + "[" + idx + "]" -> code
-    } ++ nationalAdditionalCodes.zipWithIndex.map {
-      case (code, idx) => nationalAdditionalCodesKey + "[" + idx + "]" -> code
-    }
+        case (code, idx) => taricAdditionalCodesKey + "[" + idx + "]" -> code
+      } ++ nationalAdditionalCodes.zipWithIndex.map {
+        case (code, idx) => nationalAdditionalCodesKey + "[" + idx + "]" -> code
+      }
+    )
+
+  def addActionTypeToFormData(formAction: FormAction, formData: Map[String, String]): Map[String, String] =
+    Map(formAction.label -> getKey(formAction)) ++ formData
+
+  private def getKey(action: FormAction): String = action match {
+    case Remove(keys) => keys.head
+    case _            => ""
```

Comment:
For `_` you want to have empty string? are you sure about it? I'm thinking that for `Add` you should have `Add`, but I can understand this  method wrong.

If it's only affects remove key, you can change it to `getRemovalKey`.

I'm worried about this file, because I know that we decided to keep controller tests short and small, but this one is huge. It was big before but now... Added 940 lines. I'm thinking to add some annotation, comment, TODO to this test to split it and extract a huge part of it.

Reply:
In this case, the key is the index of the element to remove. It is misleading so I changed slightly implementation to be more explanatory.